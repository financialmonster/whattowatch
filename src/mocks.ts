import { Map, fromJS, List } from 'immutable';

export const createStoreMock = (mode: string = `common`) => {
    return {
        promo: Map({
            promo: Map(),
            isPromoFetching: true,
            promoError: null
        }),
        films: Map({
            films: fromJS([{genre: `Thriller`, id: 1}, {genre: `Comedy`, id: 2}]),
            isFilmsFetching: mode === `detailed`,
            filmsError: null,
            filter: `All genres`
        }),
        auth: Map({
            user: null,
            authError: null,
            isAuthFetching: false,
            authStatusError: null,
            isAuthStatusFetching: false
        }),
        reviews: Map({
            reviews: List(),
            isReviewsFetching: true,
            reviewsError: null
        })
    }
}
