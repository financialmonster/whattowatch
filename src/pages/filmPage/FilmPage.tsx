import React, { FC } from 'react';

import { Footer } from 'components/footer/Footer';
import { DetailedFilmContainer } from 'components/detailedFilmContainer/DetailedFilmContainer';
import { SimilarFilms } from 'components/similarFilms/SimilarFilms';

export const FilmPage: FC = () => {
    return (
        <>
            <section className="movie-card movie-card--full">
                <DetailedFilmContainer />
            </section>
            <div className="page-content">
                <SimilarFilms />
                <Footer />
            </div>
        </>
    );
}
