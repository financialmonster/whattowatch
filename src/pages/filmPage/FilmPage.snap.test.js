/* eslint-disable import/first */
jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useParams: () => ({id: 1}),
    useLocation: () => ({pathname: `${process.env.PUBLIC_URL}film/1`})
}));

import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';

import { FilmPage } from './FilmPage';
import { createStoreMock } from 'mocks';

it(`FilmPage: correctly rendered`, () => {
    const mockStore = configureStore();
    const store = mockStore(createStoreMock());

    const tree = renderer.create(
        <Provider store={store}>
            <FilmPage />
        </Provider>        
    ).toJSON();

    expect(tree).toMatchSnapshot();
});
