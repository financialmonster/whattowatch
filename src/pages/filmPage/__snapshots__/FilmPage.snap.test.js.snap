// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmPage: correctly rendered 1`] = `
Array [
  <section
    className="movie-card movie-card--full"
    style={
      Object {
        "backgroundColor": undefined,
      }
    }
  >
    <div
      className="movie-card__hero"
    >
      <div
        className="movie-card__bg"
      >
        <img />
      </div>
      <h1
        className="visually-hidden"
      >
        WTW
      </h1>
      <header
        className="page-header movie-card__head"
      >
        <div
          className="logo"
        >
          <a
            className="logo__link"
            onClick={[Function]}
            style={
              Object {
                "cursor": "pointer",
              }
            }
            title="To the main page"
          >
            <span
              className="logo__letter logo__letter--1"
            >
              W
            </span>
            <span
              className="logo__letter logo__letter--2"
            >
              T
            </span>
            <span
              className="logo__letter logo__letter--3"
            >
              W
            </span>
          </a>
        </div>
        <div
          className="user-block"
        >
          <a
            className="user-block__link"
            onClick={[Function]}
            title="To the login page"
          >
            Sign in
          </a>
        </div>
      </header>
      <div
        className="movie-card__wrap"
      >
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
          />
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              Thriller
            </span>
            <span
              className="movie-card__year"
            />
          </p>
          <div
            className="movie-card__buttons"
          >
            <button
              className="btn btn--play movie-card__button"
              onClick={[Function]}
              title="Play"
              type="button"
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlinkHref="#play"
                />
              </svg>
              <span>
                Play
              </span>
            </button>
            <button
              className="btn btn--list movie-card__button"
              onClick={[Function]}
              title="My list"
              type="button"
            >
              <svg
                height="20"
                viewBox="0 0 19 20"
                width="19"
              >
                <use
                  xlinkHref="#add"
                />
              </svg>
              <span>
                My list
              </span>
            </button>
            <a
              className="btn btn--add-review movie-card__button"
              onClick={[Function]}
              title="Add review"
            >
              Add review
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      className="movie-card__wrap movie-card__translate-top"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster movie-card__poster--big"
        >
          <img
            alt="undefined poster"
            height="327"
            width="218"
          />
        </div>
        <div
          className="movie-card__desc"
        >
          <nav
            className="movie-nav movie-card__nav"
          >
            <ul
              className="movie-nav__list"
            >
              <li
                className="movie-nav__item movie-nav__item--active"
              >
                <a
                  className="movie-nav__link"
                  onClick={[Function]}
                  style={
                    Object {
                      "cursor": "default",
                    }
                  }
                  title=""
                >
                  Overview
                </a>
              </li>
              <li
                className="movie-nav__item"
              >
                <a
                  className="movie-nav__link"
                  onClick={[Function]}
                  style={Object {}}
                  title="Details"
                >
                  Details
                </a>
              </li>
              <li
                className="movie-nav__item"
              >
                <a
                  className="movie-nav__link"
                  onClick={[Function]}
                  style={Object {}}
                  title="Reviews"
                >
                  Reviews
                </a>
              </li>
            </ul>
          </nav>
          <div
            className="movie-rating"
          >
            <div
              className="movie-rating__score"
            >
              1.0
            </div>
            <p
              className="movie-rating__meta"
            >
              <span
                className="movie-rating__level"
              >
                Bad
              </span>
              <span
                className="movie-rating__count"
              >
                undefined ratings
              </span>
            </p>
          </div>
          <div
            className="movie-card__text"
          >
            <p>
              
            </p>
            <p
              className="movie-card__director"
            >
              <strong>
                Director: 
              </strong>
            </p>
            <p
              className="movie-card__starring"
            >
              <strong>
                Starring: , 
              </strong>
            </p>
          </div>
        </div>
        <div
          className="player"
          style={
            Object {
              "display": "none",
            }
          }
        >
          <div
            className="sc-AxjAm fcmpGc"
          >
            <div
              className="sc-AxirZ fLbCRQ"
              rotate="rotate(0deg)"
            />
            <div
              className="sc-AxirZ iRnakP"
              rotate="rotate(30deg)"
            />
            <div
              className="sc-AxirZ jJDLSA"
              rotate="rotate(60deg)"
            />
            <div
              className="sc-AxirZ cfISCN"
              rotate="rotate(90deg)"
            />
            <div
              className="sc-AxirZ jYaMud"
              rotate="rotate(120deg)"
            />
            <div
              className="sc-AxirZ kCauMN"
              rotate="rotate(150deg)"
            />
            <div
              className="sc-AxirZ bbuhJx"
              rotate="rotate(180deg)"
            />
            <div
              className="sc-AxirZ RfLVX"
              rotate="rotate(210deg)"
            />
            <div
              className="sc-AxirZ iDvjl"
              rotate="rotate(240deg)"
            />
            <div
              className="sc-AxirZ fchBdo"
              rotate="rotate(270deg)"
            />
            <div
              className="sc-AxirZ fGeryM"
              rotate="rotate(300deg)"
            />
            <div
              className="sc-AxirZ eVSKan"
              rotate="rotate(330deg)"
            />
          </div>
          <video
            className="player__video"
            onCanPlayThrough={[Function]}
          />
        </div>
      </div>
    </div>
  </section>,
  <div
    className="page-content"
    style={
      Object {
        "backgroundImage": "linear-gradient(-180deg, undefined 0%, undefined 100%)",
      }
    }
  >
    <section
      className="catalog catalog--like-this"
    >
      <h2
        className="catalog__title"
      >
        More like this
      </h2>
      <div
        className="catalog__movies-list"
      >
        <article
          className="small-movie-card catalog__movies-card"
        >
          <div
            className="small-movie-card__image"
            onClick={[Function]}
          >
            <img
              height="175"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
            />
          </h3>
        </article>
      </div>
    </section>
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
          title="To the main page"
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          Â© 2020 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>,
]
`;
